lib/stat_tracker.rb

```diff
@@ -1,99 +1,104 @@
-require 'csv'
-require_relative 'team_stats'
-require_relative 'game_teams_stats'
-require_relative 'game_stats'
 require_relative 'game'
+require_relative 'team'
+require_relative 'game_team'
+require_relative 'game_stats'
+require_relative 'league_stats'
+# require_relative 'season_stats'
+# require_relative 'team_stats'
+require_relative './modules/data_loadable'
 
 class StatTracker
+  attr_reader :games, :teams, :game_teams, :game_stats, :league_stats
   include DataLoadable
-  attr_reader :games_path, :team_path, :game_teams_path
 
   def self.from_csv(locations)
     StatTracker.new(locations)
   end
 
   def initialize(locations)
-    @games_path = locations[:games]
-    @team_path = locations[:teams]
-    @game_teams_path = locations[:game_teams]
-    @games_stats = GameStats.new(@games_path, Game)
-    @team_stats = TeamStats.new(@team_path, Team)
-    @game_teams_stats = GameTeamStats.new(@game_teams_path, GameTeams)
+    @games = csv_data(locations[:games], Game)
+    @teams = csv_data(locations[:teams], Team)
+    @game_teams = csv_data(locations[:game_teams], GameTeam)
+    @game_stats = GameStats.new(@games)
+    @league_stats = LeagueStats.new(@games, @teams, @game_teams)
```

Comment:
Passing in the arrays of file data  for each class as an argument is such a great way to easily access that data within the class.